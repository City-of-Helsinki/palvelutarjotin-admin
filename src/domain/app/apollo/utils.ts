import apolloClient from './apolloClient';

export const clearApolloCache = async () => {
  apolloClient.stop();
  await Promise.all([apolloClient.resetStore(), apolloClient.clearStore()]);
};
