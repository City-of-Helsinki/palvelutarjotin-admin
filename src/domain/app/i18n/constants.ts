export enum VALIDATION_MESSAGE_KEYS {
  EMAIL = 'form.validation.string.email',
  NUMBER_MAX = 'form.validation.number.max',
  NUMBER_MIN = 'form.validation.number.min',
  NUMBER_REQUIRED = 'form.validation.number.required',
  STRING_MAX = 'form.validation.string.max',
  STRING_MIN = 'form.validation.string.min',
  STRING_REQUIRED = 'form.validation.string.required',
  URL = 'form.validation.string.url',
}
