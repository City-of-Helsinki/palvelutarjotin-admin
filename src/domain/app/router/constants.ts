import type { BrowserRouterProps, MemoryRouterProps } from 'react-router-dom';

export const DEFAULT_ROUTER_PROPS = {
  future: {
    v7_startTransition: true,
  },
} as const satisfies BrowserRouterProps satisfies MemoryRouterProps;
