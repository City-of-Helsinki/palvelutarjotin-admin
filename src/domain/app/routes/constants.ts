export enum ROUTES {
  CALLBACK = '/callback',
  CREATE_EVENT = '/events/create',
  CREATE_OCCURRENCE = '/events/:id/occurrences/create',
  EDIT_OCCURRENCE = '/events/:id/occurrences/:occurrenceId/edit',
  EDIT_EVENT = '/events/:id/edit',
  ENROLLMENTS = '/events/:id/occurrences/:occurrenceId/enrollments',
  EVENT_DETAILS = '/events/:id',
  HOME = '/',
  OCCURRENCES = '/events/:id/occurrences',
  SILENT_CALLBACK = '/silent-callback',
}
