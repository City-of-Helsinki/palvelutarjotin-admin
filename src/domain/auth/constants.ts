import { LoginProviderProps } from 'hds-react';

import AppConfig from '../app/AppConfig';

// todo: delete this when new login implemented
export const API_TOKEN_ACTIONS = {
  FETCH_TOKEN_ERROR: 'FETCH_TOKEN_ERROR',
  FETCH_TOKEN_SUCCESS: 'FETCH_TOKEN_SUCCESS',
  RESET_API_TOKEN_DATA: 'RESET_API_TOKEN_DATA',
  START_FETCHING_TOKEN: 'START_FETCHING_TOKEN',
  TOKEN_FETCHED: 'TOKEN_FETCHED',
};

const providerPropertiesBase: LoginProviderProps = {
  userManagerSettings: {
    authority: AppConfig.oidcAuthority,
    client_id: AppConfig.oidcClientId,
    scope: AppConfig.oidcScope,
    response_type: AppConfig.oidcReturnType,
    redirect_uri: `${AppConfig.origin}/callback`,
    automaticSilentRenew: AppConfig.oidcAutomaticSilentRenew,
    silent_redirect_uri: `${AppConfig.origin}/silent_renew.html`,
    post_logout_redirect_uri: `${AppConfig.origin}/`,
  },
  debug: false,
  sessionPollerSettings: {
    pollIntervalInMs: AppConfig.oidcSessionPollerIntervalInMs,
  },
};

/**
 * HDS LoginProvider configuration when the Kultus Admin UI
 * is integrated with Tunnistamo.
 *
 * @deprecated LoginProviderProps for Tunnistamo integration are deprecated,
 * because the Tunnistamo instance will be changed to
 * the Helsinki-Profile Keycloak instance in the near future.
 */
export const tunnistamoPoviderProperties: LoginProviderProps = {
  ...providerPropertiesBase,
  apiTokensClientSettings: {
    url: `${AppConfig.oidcAuthority}api-tokens/`,
    maxRetries: 10,
    retryInterval: 1000,
  },
};

/**
 * HDS LoginProvider configuration when the Kultus Admin UI
 * is integrated with Keycloak.
 */
export const keycloakPoviderProperties: LoginProviderProps = {
  ...providerPropertiesBase,
  apiTokensClientSettings: {
    url: `${AppConfig.oidcAuthority}protocol/openid-connect/token`,
    maxRetries: 10,
    retryInterval: 1000,
    queryProps: {
      grantType: 'urn:ietf:params:oauth:grant-type:uma-ticket',
      permission: '#access',
    },
    audiences: AppConfig.oidcAudiences,
  },
};
