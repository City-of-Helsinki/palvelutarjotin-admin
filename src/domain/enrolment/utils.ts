import { EnrolmentFieldsFragment } from '../../generated/graphql';

export const getEnrolmentFields = (
  enrolment: EnrolmentFieldsFragment | undefined | null
) =>
  enrolment
    ? {
        extraNeeds: enrolment.studyGroup.extraNeeds,
        language:
          enrolment.person?.language || enrolment.studyGroup.person.language,
        personInfo: [
          enrolment.person?.phoneNumber,
          enrolment.person?.emailAddress,
        ].filter((e) => e),
        studyGroupPersonInfo: [
          enrolment.studyGroup.person.phoneNumber,
          enrolment.studyGroup.person.emailAddress,
        ].filter((e) => e),
      }
    : {};
