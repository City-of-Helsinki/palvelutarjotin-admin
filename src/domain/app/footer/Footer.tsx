import { Footer as HDSFooter, Logo, logoFi, logoSv } from 'hds-react';
import React, { FunctionComponent } from 'react';
import type { MenuItem } from 'react-helsinki-headless-cms';
import { useMenuQuery } from 'react-helsinki-headless-cms/apollo';
import { useTranslation } from 'react-i18next';

import { resetFocusId } from '../../../common/components/resetFocus/ResetFocus';
import { FOOTER_MENU_NAME } from '../../../headless-cms/constants';
import useLocale from '../../../hooks/useLocale';
import styles from './footer.module.scss';

type ValidMenuItem = {
  id: MenuItem['id'];
  label: NonNullable<MenuItem['label']>;
  path: NonNullable<MenuItem['path']>;
};

const isValidMenuItem = (menuItem: MenuItem): menuItem is ValidMenuItem =>
  !!menuItem.label && !!menuItem.path;

const Footer: FunctionComponent = () => {
  const { t } = useTranslation();
  const locale = useLocale();

  // override Footer component default behaviour which focuses skip-link
  const handleBackToTop = () => {
    window?.scrollTo({ top: 0 });
    document.querySelector<HTMLDivElement>(`#${resetFocusId}`)?.focus();
  };

  const footerMenuQuery = useMenuQuery({
    variables: { id: FOOTER_MENU_NAME[locale], menuIdentifiersOnly: true },
  });
  const footerLinks =
    footerMenuQuery.data?.menu?.menuItems?.nodes?.filter(isValidMenuItem) ?? [];

  return (
    <HDSFooter title={t('appName')} className={styles.footer}>
      <HDSFooter.Base
        copyrightHolder={t('footer.copyright')}
        copyrightText={t('footer.allRightsReserved')}
        logo={
          <Logo
            src={locale === 'sv' ? logoSv : logoFi}
            size="medium"
            alt={t('common.cityOfHelsinki')}
          />
        }
        backToTopLabel={t('footer.backToTop')}
        onBackToTopClick={handleBackToTop}
      >
        {footerLinks.map((footerLink) => (
          <HDSFooter.Link
            key={footerLink.id}
            href={footerLink.path}
            label={footerLink.label}
          />
        ))}
      </HDSFooter.Base>
    </HDSFooter>
  );
};

export default Footer;
