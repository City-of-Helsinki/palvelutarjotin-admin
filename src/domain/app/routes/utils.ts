import { ROUTES } from './constants';

export const getCmsPath = (slug?: string | null) => {
  return ROUTES.CMS_PAGE.replace('/:slug', slug ?? '');
};
