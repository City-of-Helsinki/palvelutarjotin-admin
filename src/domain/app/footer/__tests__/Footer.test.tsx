import * as React from 'react';
import { MockedResponse } from '@apollo/client/testing';
import {
  LanguagesDocument,
  MenuDocument,
} from 'react-helsinki-headless-cms/apollo';
import { vi } from 'vitest';

import Footer from '../Footer';
import { render, screen } from '../../../../utils/testUtils';
import { FOOTER_MENU_NAME } from '../../../../headless-cms/constants';
import languagesResponse from '../../../../test/apollo-mocks/queryResponses/languages.json';
import footerMenuResponse from '../../../../test/apollo-mocks/queryResponses/footerMenu.json';

vi.mock('hds-react', async () => {
  const actual = await vi.importActual('hds-react');
  return {
    ...actual,
    logoFi: 'mocked hds-react logoFi',
  };
});

const mocks: MockedResponse[] = [
  {
    request: {
      query: MenuDocument,
      variables: {
        id: FOOTER_MENU_NAME['fi'],
        menuIdentifiersOnly: true,
      },
    },
    result: footerMenuResponse,
  },
  {
    request: {
      query: LanguagesDocument,
    },
    result: languagesResponse,
  },
];

it('Footer matches snapshot', async () => {
  const { container } = render(<Footer />, { mocks });
  await screen.findByText('Kouluille ja päiväkodeille');
  expect(container.firstChild).toMatchSnapshot();
});
