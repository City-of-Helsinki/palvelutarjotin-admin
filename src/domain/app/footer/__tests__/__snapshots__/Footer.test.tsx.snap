// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches snapshot 1`] = `"<footer class=\\"footerWrapper\\"><div class=\\"Koros-module_koros__1r3rg koros\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"85\\"><defs><pattern id=\\"koros_basic-1\\" x=\\"0\\" y=\\"0\\" width=\\"106\\" height=\\"85\\" patternUnits=\\"userSpaceOnUse\\"><path transform=\\"scale(5.3)\\" d=\\"M0,800h20V0c-4.9,0-5,2.6-9.9,2.6S5,0,0,0V800z\\"></path></pattern></defs><rect fill=\\"url(#koros_basic-1)\\" width=\\"100%\\" height=\\"85\\"></rect></svg></div><div class=\\"container default\\"><div class=\\"footer\\"><div class=\\"helsinkiLogo\\"></div><div class=\\"copyright\\"><p>© Copyright 2020 • Kaikki oikeudet pidätetään</p></div></div></div></footer>"`;
